src/aap_eda/services/validation.py

```diff
@@ -0,0 +1,97 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import urllib.parse
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+
+from aap_eda.core import models
+from aap_eda.services.ruleset.activate_rulesets import ACTIVATION_PATH
+
+
+def validate_activation(
+    decision_environment_id: int,
+    rulebook_id: int,
+    user_id: int,
+    extra_var_id: int,
+) -> None:
+    _validate_awx_token(user_id)
+    _validate_decision_environment(decision_environment_id)
+    _validate_extra_var(extra_var_id)
+    _validate_rulebook(rulebook_id)
+    _validate_websocket()
+
+
+def _validate_awx_token(user_id) -> None:
+    tokens = models.AwxToken.objects.filter(user_id=user_id).count()
+    if tokens == 0:
+        raise ValidationError(_("No controller token specified"))
+    elif tokens > 1:
+        raise ValidationError(
+            _(
+                "More than one controller token found, "
+                "currently only 1 token is supported"
+            )
+        )
+
+
+def _validate_websocket() -> None:
```

Comment:
I think this check is wrong.  We are rejecting a valid request for a value that is set in the server and the user has no control. This would be a server error, not a client error. This check should follow a different flow and returns a 500 response